File: templates/service-java/{{#server.spring}}/src/main/java/{{package.value}}/{{#jsonlib.jackson}}/ObjectMapperConfig.java
package {{package.value}};

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.*;
import {{package.value}}.json.*;

@Configuration
public class ObjectMapperConfig {
	@Bean
	@Primary
	public ObjectMapper objectMapper() {
		var objectMapper = new ObjectMapper();
		CustomObjectMapper.setup(objectMapper);
		return objectMapper;
	}

    @Bean
    public Json json() {
        return new Json(objectMapper());
    }
}

File: templates/service-kotlin/{{#server.spring}}/src/main/kotlin/{{package.value}}/{{#jsonlib.jackson}}/ObjectMapperConfig.kt
package {{package.value}}

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.springframework.context.annotation.*
import {{package.value}}.json.*

@Configuration
class ObjectMapperConfig {
    @Bean
    @Primary
    fun objectMapper(): ObjectMapper {
        val objectMapper = jacksonObjectMapper()
        setupObjectMapper(objectMapper)
        return objectMapper
    }

    @Bean
    fun json(): Json {
        return Json(objectMapper())
    }
}
